app/gui/src/project-view/providers/interactionHandler.ts

```diff
@@ -26,15 +29,38 @@ export class InteractionHandler {
     })
   }
 
-  /** Set current interactoin. Any existing interaction will be ended (not cancelled). */
+  /**
+   * Same as {@link setWhen}, but provides a callback which will be called with the currently active interaction.
+   *
+   * It is roughly the same as `setWhen(active, interactionBuider(handler.getCurrent()))`.
+   */
+  setWhenWithParent(
+    active: WatchSource<boolean>,
+    interactionBuilder: (parent: Interaction | undefined) => Interaction,
+  ) {
+    const activeInteraction = ref<Interaction>()
```

Comment:
I think an `onCleanup` in the `watch` could make `activeInteraction` unnecessary.

Reply:
Yep, fixed